testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -11262,11 +11264,12 @@ private static void checkSubstringFunction(SqlOperatorFixture f) {
           String.format(Locale.ROOT, "^substring('string', CAST(%d AS DOUBLE), "
               + "CAST(%d AS DOUBLE))^", Byte.MIN_VALUE, Byte.MAX_VALUE + 10),
           "Cannot apply 'SUBSTRING' to arguments of type "
-            + ".*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*", false);
+            + ".*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*\\n.*", false);
```

Comment:
What changed here?

Reply:
Because of implicit casts SUBSTRING can now also allow UUID as arguments, which are implicitly cast to VARCHAR.